src/components/FeatureChecking/withFeatureProvider.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+import * as React from 'react';
+import FeatureProvider from './FeatureProvider';
+import type { FeatureConfig } from './flowTypes';
+
+type WrapperProps = {
+    features: FeatureConfig,
+};
+
+function withFeatureProvider(WrappedComponent: React.ComponentType<any>) {
+    function WrapperComponent({ features, ...props }: WrapperProps) {
+        return (
+            <FeatureProvider features={features}>
+                <WrappedComponent {...props} />
+            </FeatureProvider>
+        );
+    }
+    const wrappedName = WrappedComponent.displayName || WrappedComponent.name || '';
+    WrapperComponent.displayName = `withFeatureProvider(${wrappedName})`;
+    return WrapperComponent;
+}
+
+export default withFeatureProvider;
```

Comment:
Would this be more accurately named `withFeaturesProvider`, plural, since it injects the `features` prop?

Reply:
well... it injects the FeatureProvider which takes a features prop 🤷‍♂️